src/router/routes/genericResourceRoute.ts

```diff
@@ -36,7 +36,12 @@ export default class GenericResourceRoute {
                     // Get the ResourceType looks like '/Patient'
                     const resourceType = req.baseUrl.substr(1);
                     const { id } = req.params;
-                    const response = await this.handler.read(resourceType, id);
+                    let response = await this.handler.read(resourceType, id);
+                    response = this.authService.authorizeAndFilterReadResponse({

```

Comment:
Nitpick: Instead of reusing the response variable, I would rename the variable on this line to be something along the lines of `authorizedAndFilteredReadResponse` or `updatedReadResponse`.